### PR TITLE
Cherry-pick #8653 to 6.x: Fix a race condition on add_host_metadata

### PR DESCRIPTION
**Pull request by jsoriano**
Cherry-pick of PR #8653 to 6.x branch. Original message: 

Not sure if adding a mutex at this level will be too blocking for events processing, I have also though on other options to reduce contention on reads:
* Using a RWLock, but we'd still need to write-lock first to check expiration and update
* Using a RWLock and a go-routine to do periodic updates, but afaik we wouldn't have a way to stop it if the processor is removed.

Continues with #8223

### CODE REVIEW

**Comment by jsoriano** [comment]
jenkins, test this again please

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -40,7 +40,7 @@ https://github.com/elastic/beats/compare/v6.4.0...6.x[Check the HEAD diff]
 - Deregister pipeline loader callback when inputsRunner is stopped. {pull}[7893][7893]
 - Add backoff support to x-pack monitoring outputs. {issue}7966[7966]
 - Removed execute permissions systemd unit file. {pull}7873[7873]
-- Fix a race condition with the `add_host_metadata` and the event serialization. {pull}8223[8223]
+- Fix a race condition with the `add_host_metadata` and the event serialization. {pull}8223[8223] {pull}8653[8653]
```
Comment (sayden) [inline CHANGELOG.asciidoc]: Double pull? What does it mean?

Comment (jsoriano) [reply CHANGELOG.asciidoc]: This change is following another related fix (#8223), so both PRs are about the same issue.

Comment (sayden) [reply CHANGELOG.asciidoc]: Thanks for the clarification :slightly_smiling_face:

**Pull request merged**